azurelinuxagent/common/utils/shellutil.py

```diff
@@ -325,4 +337,50 @@ def quote(word_list):
 
     return " ".join(list("'{0}'".format(s.replace("'", "'\\''")) for s in word_list))
 
-# End shell command util functions
+
+#
+# The run_command/run_pipe/run/run_get_output functions use these callbacks to notify when a command has been started
+# and when it completes.
+#
+# The callbacks are executed in the context of the thread executing run_command/run_pipe/run/run_get_output so they
+# should be used only to perform very fast operations or otherwise would block that thread. Also, errors in the
+# callback are ignored, so they should do they own error handling.
+#
+# Only 1 callback is maintained for each of those notifications. Setting a callback overrides the previous value.
+# Set the callback to None to stop the notifications
+#
+# C0103: Constant name "foo" doesn't conform to UPPER_CASE naming style (invalid-name) -- Disabled: These are not constants
+_on_command_started_callback = None  # pylint:disable=C0103
+_on_command_completed_callback = None  # pylint:disable=C0103
+
+
+def set_on_command_started_callback(callback):
```

Comment:
This is not used anywhere? Maybe I'm missing something, is this only the first PR of multiple that involves callback changes?

If I'm seeing this correctly, `_invoke_on_command_started_callback` does nothing since the `_on_command_started_callback` will always be None, correct?

Reply:
It will be used to check the processes in the agent cgroup